src/du/du.rs

```diff
@@ -75,8 +76,76 @@ impl Stat {
     }
 }
 
-fn get_default_blocks() -> u64 {
-    1024
+fn translate_to_pure_number(s: &str) -> Option<u64> {
+    let mut found_number = false;
+    let mut found_letter = false;
+    let mut numbers = String::new();
+    let mut letters = String::new();
+    for c in s.trim().chars() {
+        if found_letter && c.is_digit(10) || !found_number && !c.is_digit(10) {
+            return None;
+        } else if c.is_digit(10) {
+            found_number = true;
+            numbers.push(c);
+        } else if c.is_alphabetic() {
+            found_letter = true;
+            letters.push(c);
+        }
+    }
+    let number = numbers.parse::<u64>().unwrap();
+    let multiple = match letters[..].to_uppercase().as_str() {
+        "" => 1,
+        "K" => 1024u64.pow(1),
+        "M" => 1024u64.pow(2),
+        "G" => 1024u64.pow(3),
+        "T" => 1024u64.pow(4),
+        "P" => 1024u64.pow(5),
+        "E" => 1024u64.pow(6),
+        "Z" => 1024u64.pow(7),
+        "Y" => 1024u64.pow(8),
+        "KB" => 1000u64.pow(1),
+        "MB" => 1000u64.pow(2),
+        "GB" => 1000u64.pow(3),
+        "TB" => 1000u64.pow(4),
+        "PB" => 1000u64.pow(5),
+        "EB" => 1000u64.pow(6),
+        "ZB" => 1000u64.pow(7),
+        "YB" => 1000u64.pow(8),
+        _ => return None,
+    };
+    Some(number * multiple)
+}
+
+fn translate_to_pure_number_from_option(s: &Option<String>) -> Option<u64> {
```

Comment:
Sorry, I didn't notice that this was called twice for some reason.  I prefer the way it was before.